file-server/src/main/java/edu/lysak/fileserver/server/FileStorage.java

```diff
@@ -1,26 +1,81 @@
 package edu.lysak.fileserver.server;
 
+import edu.lysak.fileserver.util.SerializationUtils;
+
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.security.SecureRandom;
+import java.util.HashMap;
+import java.util.Map;
+
+import static edu.lysak.fileserver.util.Constants.SERVER_STORAGE_FOLDER;
 
 public class FileStorage {
-//    private static final String STORAGE_FOLDER = System.getProperty("user.dir") + "/File Server/task/src/server/data/";
-    private static final String STORAGE_FOLDER = "file-server/src/main/resources/server/data/";
+    private Map<Integer, String> identifiers;
+
+    public int addFile(String fileName, byte[] fileBytes) throws IOException {
+        int id = getId();
+        identifiers.put(id, fileName);
+        updateIdentifiersFile();
+        Files.write(Path.of(SERVER_STORAGE_FOLDER + fileName), fileBytes);
+        return id;
+    }
+
+    public byte[] getFileById(int id) throws IOException {
+        String fileName = identifiers.get(id);
+        Path filePath = Path.of(SERVER_STORAGE_FOLDER + fileName);
+        if (fileName != null && Files.exists(filePath)) {
+            return Files.readAllBytes(filePath);
+        }
+        return null;
+    }
 
-    public void addFile(String fileName, String fileContent) throws IOException {
-        Files.writeString(Path.of(STORAGE_FOLDER + fileName), fileContent);
+    public byte[] getFileByName(String fileName) throws IOException {
+        System.out.println("FILE_NAME = " + fileName);
+        Path filePath = Path.of(SERVER_STORAGE_FOLDER + fileName);
+        if (Files.exists(filePath)) {
+            return Files.readAllBytes(filePath);
+        }
+        return null;
     }
 
-    public String getFileContent(String fileName) throws IOException {
-        return new String(Files.readAllBytes(Path.of(STORAGE_FOLDER + fileName)));
+    public boolean deleteFileById(int id) throws IOException {
+        String fileName = identifiers.remove(id);
+        boolean isDeleted = Files.deleteIfExists(Path.of(SERVER_STORAGE_FOLDER + fileName));
+        updateIdentifiersFile();
+        return isDeleted;
     }
 
-    public boolean deleteFile(String fileName) throws IOException {
-        return Files.deleteIfExists(Path.of(STORAGE_FOLDER + fileName));
+    public boolean deleteFileByName(String fileName) throws IOException {
+        boolean isDeleted = Files.deleteIfExists(Path.of(SERVER_STORAGE_FOLDER + fileName));
+        updateIdentifiersFile();
+        return isDeleted;
     }
 
     public boolean contains(String fileName) {
-        return Files.exists(Path.of(STORAGE_FOLDER + fileName));
+        return Files.exists(Path.of(SERVER_STORAGE_FOLDER + fileName));
+    }
+
+    private int getId() {
+        int id;
+        do {
+            id = new SecureRandom().nextInt(1000);
+        } while (identifiers.containsKey(id));
+        return id;
+    }
```

Comment:
Think about UUID.

Reply:
It is not applicable for the task requirements